### PR TITLE
Intacct app redesign : landing page, services, models/enums

### PR DESCRIPTION
**Pull request by anishfyle**
https://github.com/fylein/fyle-integrations-settings-app/assets/116036738/d1690e00-9551-4326-8de1-21271b2cf536

### CODE REVIEW

File: src/app/integrations/integrations-routing.module.ts [hunk 1]
```diff
@@ -27,6 +27,10 @@ const routes: Routes = [
       {
         path: 'gusto',
         loadChildren: () => import('./gusto/gusto.module').then(m => m.GustoModule)
+      },
+      {
+        path: 'si',
```
Comment (ashwin1111) [inline src/app/integrations/integrations-routing.module.ts:32]: intacct

File: src/app/integrations/landing/landing.component.ts [hunk 2]
```diff
@@ -53,7 +53,8 @@ export class LandingComponent implements OnInit {
     [InAppIntegration.BAMBOO_HR]: '/integrations/bamboo_hr/',
     [InAppIntegration.QBD]: '/integrations/qbd/',
     [InAppIntegration.TRAVELPERK]: '/integrations/travelperk/',
-    [InAppIntegration.GUSTO]: '/integrations/gusto/'
+    [InAppIntegration.GUSTO]: '/integrations/gusto/',
+    [InAppIntegration.SI]: '/integrations/si/'
```
Comment (ashwin1111) [inline src/app/integrations/landing/landing.component.ts:57]: intacct

File: src/app/integrations/si/onboarding/onboarding-landing/onboarding-landing.component.html [hunk 3]
```diff
@@ -1 +1,8 @@
-<p>onboarding-landing works!</p>
+<div>
+    <div>
+        <app-app-header [iconPath]="'assets/logos/sage-intacct-logo.svg'" [appName]="'Sage Intacct'" [appDescription]="'Import data from Sage Intacct to Fyle and Export expenses from Fyle to Sage Intacct. '" [isLoading]="isLoading" [isIntegrationSetupInProgress]="isSISetupInProgress" [isIntegrationConnected]="isSIConnected"></app-app-header>
+    </div>
+    <div>
+        <app-app-integration-landing [headerText]="'A quick guide to help you set up the integration quick and easy.'" [showVideo] = "true"[svgPath]="'assets/flow-charts/sage-flow-chart.svg'" [redirectLink]="RedirectLink.SI"></app-app-integration-landing>
```
Comment (ashwin1111) [inline src/app/integrations/si/onboarding/onboarding-landing/onboarding-landing.component.html:6]: in enum, EMBED_LINK = {INTACCT: 'https://asdasdasd.asdad.asd'}
[embedLink]="EMBED_LINK.INTACCT"

File: src/app/integrations/si/onboarding/onboarding-landing/onboarding-landing.component.ts [hunk 4]
```diff
@@ -7,6 +8,14 @@ import { Component, OnInit } from '@angular/core';
 })
 export class OnboardingLandingComponent implements OnInit {
 
+  RedirectLink = RedirectLink;
+
+  isSISetupInProgress: boolean = false;
+
+  isSIConnected: boolean = false;
+
+  isLoading: boolean = false;
```
Comment (ashwin1111) [inline src/app/integrations/si/onboarding/onboarding-landing/onboarding-landing.component.ts:17]: if the values are not gonna change, directly hard code it to the .html

File: src/app/integrations/si/si.component.ts [hunk 5]
```diff
@@ -7,6 +12,14 @@ import { Component, OnInit } from '@angular/core';
 })
 export class SiComponent implements OnInit {
 
+  user: MinimalUser;
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:15]: add workspace creation / get operations and redirect it to required page

File: src/app/shared/components/apps/app-header/app-header.component.ts [hunk 6]
```diff
@@ -68,6 +68,11 @@ export class AppHeaderComponent implements OnInit {
     this.router.navigate(['/integrations/qbd/onboarding/export_settings']);
   }
 
+  connectSI(): void {
+    this.trackingService.onClickEvent(ClickEvent.CONNECT_QBD);
+    this.router.navigate(['/integrations/si/onboarding/export_settings']);
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.ts:73]: Get route from parent component, ex - `si/onboarding/export_settings`, make the function generic

File: src/app/shared/components/apps/app-header/app-header.component.html [hunk 7]
```diff
@@ -37,6 +37,10 @@
                     Start
                     <i class="tw-pl-10-px tw-pt-2-px !tw-text-12-px pi pi-arrow-right"></i>
                 </button>
+                <button *ngIf="appName === AppName.SI" pButton type="button" class="p-button-raised" (click)="connectSI()">
+                    Start
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.html:41]: get button text from parent component, re use 37th line, update ng if to app == qbd/si

File: src/app/shared/components/apps/app-integration-landing/app-integration-landing.component.html [hunk 8]
```diff
@@ -8,6 +8,10 @@
         </span>
     </p>
 
+    <div *ngIf = 'showVideo' class="tw-flex tw-justify-center tw-pt-48-px">
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-integration-landing/app-integration-landing.component.html:11]: use double quote always, no space before and after =

File: src/app/shared/components/apps/app-integration-landing/app-integration-landing.component.ts [hunk 9]
```diff
@@ -13,6 +13,7 @@ export class AppIntegrationLandingComponent implements OnInit {
 
   @Input() redirectLink: string;
 
+  @Input() showVideo: boolean;
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-integration-landing/app-integration-landing.component.ts:16]: change it to embedLink: ENUM

File: src/app/shared/components/apps/app-header/app-header.component.ts (hunk 6 above)
Comment (anishfyle) [reply src/app/shared/components/apps/app-header/app-header.component.ts:73]: resolved in branched PR

File: src/app/integrations/si/si.component.ts (hunk 5 above)
Comment (anishfyle) [reply src/app/integrations/si/si.component.ts]: resolved in branched PR

File: src/app/core/models/db/workspaces.model.ts [hunk 10]
```diff
@@ -0,0 +1,20 @@
...
+}
+
+export interface IntacctWorkspace extends Workspace {
+  last_synced_at: Date;
+  ccc_last_synced_at: Date;
+  source_synced_at: Date;
+  cluster_domain: string;
+  destination_synced_at: Date;
+  onboarding_state: IntacctOnboardingState;
+}
```
Comment (ashwin1111) [inline src/app/core/models/db/workspaces.model.ts:20]: move this to intacct wala folder

File: src/app/core/models/enum/enum.model.ts [hunk 11]
```diff
@@ -80,14 +87,25 @@ export enum ClickEvent {
...
+  CONNECT_SI  = 'Connect Sage Intacct'
 }
 
 export enum ProgressPhase {
   ONBOARDING = 'Onboarding',
   POST_ONBOARDING = 'Post Onboarding'
 }
 
+export enum IntacctOnboardingState {
+  LANDING = 'LANDING',
```
Comment (ashwin1111) [inline src/app/core/models/enum/enum.model.ts:100]: don't think this is present

File: src/app/core/models/misc/onboarding-stepper.model.ts [hunk 12]
```diff
@@ -0,0 +1,12 @@
+export type OnboardingStepper = {
```
Comment (ashwin1111) [inline src/app/core/models/misc/onboarding-stepper.model.ts:1]: use the same everywhere, if we have separate model for qbd, remove the model and replace imports

File: src/app/core/services/si/si-core/si-workspace.service.ts [hunk 13]
```diff
@@ -1,9 +1,41 @@
...
+  postWorkspace(): Observable<IntacctWorkspace> {
+    return this.apiService.post('/workspaces/', {});
+  }
+
+  getWorkspaceId(): string {
+    return this.storageService.get('si.workspaceId');
+  }
+
+  setIntacctOnboardingState(onboardingState: IntacctOnboardingState): void {
+    return this.storageService.set('IntacctOnboardingState', onboardingState);
```
Comment (ashwin1111) [inline src/app/core/services/si/si-core/si-workspace.service.ts:33]: `si.onboardingState`

File: src/app/core/services/si/si-core/si-workspace.service.ts [hunk 14]
```diff
@@ -1,9 +1,41 @@
...
+  }
+
+  getIntacctOnboardingState(): IntacctOnboardingState {
+    const onboardingState = this.storageService.get('IntacctOnboardingState');
```
Comment (ashwin1111) [inline src/app/core/services/si/si-core/si-workspace.service.ts:37]: `si.onboardingState`

File: src/app/core/services/si/si-core/si-workspace.service.ts [hunk 15]
```diff
@@ -1,9 +1,41 @@
...
+    return onboardingState ? onboardingState : IntacctOnboardingState.LANDING;
```
Comment (ashwin1111) [inline src/app/core/services/si/si-core/si-workspace.service.ts:38]: change default state to connection

File: src/app/integrations/si/onboarding/onboarding-routing.module.ts [hunk 16]
```diff
@@ -1,12 +1,46 @@
...
 
+const routes: Routes = [
+  {
+    path: '',
+    component: OnboardingComponent,
+    children: [
+      {
+        path: 'landing',
+        component: OnboardingLandingComponent
+      },
```
Comment (ashwin1111) [inline src/app/integrations/si/onboarding/onboarding-routing.module.ts:21]: add 1 more route for intacct_connector

Comment (ashwin1111) [reply src/app/integrations/si/onboarding/onboarding-routing.module.ts:21]: name it just connector

File: src/app/integrations/si/si.component.ts [hunk 17]
```diff
@@ -7,9 +15,68 @@ import { Component, OnInit } from '@angular/core';
...
+      this.workspaceService.postWorkspace().subscribe((workspaces: any) => {
+        this.workspaceSetting(workspaces);
+      });
+    }
+    );
+  }
+
+  setupWorkspace() {
+    this.user = this.userService.getUserProfile();
+      this.getOrCreateWorkspace();
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:75]: indent

File: src/app/integrations/si/si.component.ts [hunk 18]
```diff
@@ -7,9 +15,68 @@ import { Component, OnInit } from '@angular/core';
 })
 export class SiComponent implements OnInit {
 
-  constructor() { }
+  user: MinimalUser;
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:18]: = this.userService.getUserProfile();

File: src/app/integrations/si/si.component.ts [hunk 19]
```diff
@@ -7,9 +15,68 @@ import { Component, OnInit } from '@angular/core';
...
+    }, (error) => {
+      this.workspaceService.postWorkspace().subscribe((workspaces: any) => {
+        this.workspaceSetting(workspaces);
+      });
+    }
+    );
+  }
+
+  setupWorkspace() {
+    this.user = this.userService.getUserProfile();
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:74]: move to initialiser

File: src/app/integrations/si/si.component.ts [hunk 20]
```diff
@@ -7,9 +15,68 @@ import { Component, OnInit } from '@angular/core';
...
+        [IntacctOnboardingState.EXPORT_SETTINGS]: '/integrations/intacct/onboarding/export_settings',
+        [IntacctOnboardingState.IMPORT_SETTINGS]: '/integrations/intacct/onboarding/import_settings',
+        [IntacctOnboardingState.ADVANCED_SETTINGS]: '/integrations/intacct/onboarding/advanced_settings',
+        [IntacctOnboardingState.COMPLETE]: '/integrations/intacct/main'
+      };
+      this.router.navigateByUrl(onboardingStateComponentMap[this.workspace.onboarding_state]);
+    }
+  }
+
+  workspaceSetting(workspace:IntacctWorkspace) {
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:52]: setupWorkspace()

File: src/app/integrations/si/si.component.ts [hunk 21]
```diff
@@ -7,9 +15,68 @@ import { Component, OnInit } from '@angular/core';
...
+    this.workspace = workspace;
+    this.storageService.set('si.workspaceId', this.workspace.id);
+    this.storageService.set('IntacctOnboardingState', this.workspace.onboarding_state);
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:55]: key name change

File: src/app/integrations/si/si.component.ts [hunk 22]
```diff
@@ -7,9 +15,68 @@ import { Component, OnInit } from '@angular/core';
...
+    private windowService: WindowService
+  ) {
+    this.windowReference = this.windowService.nativeWindow;
+  }
+
+  private navigate(): void {
+    const pathName = this.windowReference.location.pathname;
+    if (pathName === '/integrations/intacct') {
+      const onboardingStateComponentMap = {
+        [IntacctOnboardingState.LANDING]: '/integrations/intacct/onboarding/landing',
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:40]: remove

File: src/app/integrations/si/si.component.ts [hunk 23]
```diff
@@ -7,9 +15,68 @@ import { Component, OnInit } from '@angular/core';
...
+        [IntacctOnboardingState.CONNECTION]: '/integrations/intacct/onboarding/connection',
+        [IntacctOnboardingState.LOCATION_ENTITY]: '/integrations/intacct/onboarding/location_entity',
```
Comment (ashwin1111) [inline src/app/integrations/si/si.component.ts:42]: for location_entity we should redirect to connector/

File: src/app/shared/components/apps/app-header/app-header.component.html [hunk 24]
```diff
@@ -33,9 +33,9 @@
                 <button *ngIf="isIntegrationConnected && isRecipeRunning && appName !== AppName.GUSTO" pButton type="button" class="!tw-text-alert-toast remove-bg disconnect !tw-bg-white" (click)="disconnect()">
                     Disconnect
                 </button>
-                <button *ngIf="appName === AppName.QBD" pButton type="button" class="p-button-raised" (click)="connectQBD()">
-                    Start
-                    <i class="tw-pl-10-px tw-pt-2-px !tw-text-12-px pi pi-arrow-right"></i>
+                <button *ngIf="appName === AppName.QBD" pButton type="button" class="p-button-raised" (click)="connectIntegration()">
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.html:36]: || intacct

File: src/app/shared/components/apps/app-header/app-header.component.html [hunk 25]
```diff
@@ -33,9 +33,9 @@
...
+                    {{buttonText}}
+                    <i *ngIf="appName === AppName.QBD" class="tw-pl-10-px tw-pt-2-px !tw-text-12-px pi pi-arrow-right"></i>
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.html:38]: here too

File: src/app/shared/components/apps/app-header/app-header.component.ts [hunk 26]
```diff
@@ -68,6 +70,19 @@ export class AppHeaderComponent implements OnInit {
     this.router.navigate(['/integrations/qbd/onboarding/export_settings']);
   }
 
+  connectSI(): void {
+    this.trackingService.onClickEvent(ClickEvent.CONNECT_QBD);
+    this.router.navigate(['/integrations/si/onboarding/landing']);
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.ts:75]: get `@Input() postConnectionRoute: string` which will have values like `si/onboarding/landing` or `qbd/onboarding/landing`. And use the variable for navigate()

File: src/app/shared/components/apps/app-header/app-header.component.ts [hunk 27]
```diff
@@ -68,6 +70,19 @@ export class AppHeaderComponent implements OnInit {
...
+  }
+
+  connectIntegration(): void {
+    if (this.appName === AppName.QBD) {
+      this.connectQBD();
+    } else if (this.appName === AppName.INTACCT) {
+      this.connectSI;
+    }
+  }
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.ts:84]: not needed

File: src/app/shared/components/apps/app-header/app-header.component.ts [hunk 28]
```diff
@@ -68,6 +70,19 @@ export class AppHeaderComponent implements OnInit {
     this.router.navigate(['/integrations/qbd/onboarding/export_settings']);
   }
 
+  connectSI(): void {
```
Comment (ashwin1111) [inline src/app/shared/components/apps/app-header/app-header.component.ts:73]: name it generic

File: src/app/shared/components/apps/app-header/app-header.component.html (hunk 25 above)
Comment (anishfyle) [reply src/app/shared/components/apps/app-header/app-header.component.html:38]: we wont need arrow icon in intacct so will pass this

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**